docs/src/references/agent/operation-workflow.md

```diff
@@ -328,6 +332,56 @@ Note that:
 - No `on_exit` nor `on_kill` status can be provided, as the script is not monitored.
 - If the script cannot be launched, the workflow will be moved to the final `"failed"` state.
 
+### Sub-Command Execution
+
+An operation workflow can trigger a command defined by another workflow.
+
+```toml
+["<state-name>"]
+command = "<operation-name>"
+```
+
+Operation specific arguments can be provided to the sub-command instance,
+and data extracted from the calling command status topic and message payload can be injected into these arguments.
+
+- For instance, `cmd_arg.url = "http://127.0.0.1:8000/tedge/file-transfer/foo.txt"`
+  adds an `url` property to the init message of the sub-command.
+- Similarly, `cmd_arg.url = "${.payload.config_update.url}"`
+  adds an `url` property to the init message of the sub-command, but extracting this url from the calling command state.
+- Several such arguments can be added to the sub-command initial state.
+  - All these individual arguments are combined into a JSON object value published on the sub-command MQTT topic.
+  - The sub-command initial state can be provided as a whole with `cmd_arg = ${.payload.path.to.sub-command.init-status}`
+  - The calling command excerpts injected into this initial state are defined using the same path conventions as for script handlers.
+  - If a `cmd_arg.status` value is provided, this value will be ignored and replaced by `"init"`
+    as this is the only valid value for the initial state of a command.
+  - If no value at all is provided for `cmd_arg`, then the default value is `{"status" = "init"}`.
+- There is no way to assign a command identifier for the new sub-command instance.
+  - This identifier is generated by %%te%%, which creates, monitors and clears the MQTT topic for the sub-command.
+
+From an execution perspective, triggering a sub-command is similar to running a background script.
+
+- The calling command workflow is paused while the sub-command is processed.
+- The calling workflow move to an `on_exec` state from where to sub-command execution is watched till success, failure or timeout.
+- A specific `"await-command-completion"` action is to be used for this `on_exec` state where the sub-command completion is awaited.
+- The steps following the sub-command completion are defined on the `on_exec` state with `on_success`, `on_error` and `on_timeout` handlers.
+
+For example, a configuration update can be triggered with:
+
+```toml
+[trigger_config_update]
+command = "config_update"
+cmd_arg.tedgeUrl = "http://127.0.0.1:8000/tedge/file-transfer/example/config_update/mosquitto-1234",
+cmd_arg.type = "mosquitto"
+on_exec = "waiting_for_config_update"
+
+[waiting_for_config_update]
+action = "await-command-completion"
```

Comment:
Having two states and a new `await-command-completion` is required from a technical point of view. However, this might seem a bit contrived  from a user point of view. What I propose is to get first a working solution and only then to add some syntactic sugar to let the user define a single state with all the command related handlers and to let the system introduce the waiting state.